pkg/controllers/job/job_controller.go

```diff
@@ -203,6 +206,15 @@ func (cc *Controller) processNextReq() bool {
 		return true
 	}
 
+	job, err := cc.jobLister.Jobs(req.Namespace).Get(req.JobName)
```

Comment:
there's maybe multiple requests for a job, which will make it have different workers handling the same thing.

Reply:
Yes, it may occur, for example when multi-pods event arrives.There is a risk for  causing job.status outdated.  

The safe way is to add a lock not to handle same job concurrently.